leakpro/leakpro.py

```diff
@@ -1,22 +1,52 @@
 """Main class for LeakPro."""
 
+import types
 from pathlib import Path
 
 import yaml
 
 from leakpro.attacks.attack_scheduler import AttackScheduler
 from leakpro.reporting.utils import prepare_privacy_risk_report
 from leakpro.user_inputs.abstract_input_handler import AbstractInputHandler
+from leakpro.user_inputs.handler_setup import (
+    _load_model_class,
+    _load_population,
+    _load_target_metadata,
+    _load_trained_target_model,
+    _validate_indices,
+    _validate_target_metadata,
+    get_dataloader,
+    get_dataset,
+    get_labels,
+    get_population_size,
+    get_target_model,
+    get_target_model_blueprint,
+    get_target_model_metadata,
+    get_target_replica,
+    get_test_indices,
+    get_train_indices,
+    set_target_model,
+    set_target_model_blueprint,
+    set_target_model_metadata,
+    set_test_indices,
+    set_train_indices,
+    setup,
+)
+from leakpro.user_inputs.modality_extensions.tabular_extension import TabularExtension
 from leakpro.utils.import_helper import Self
 from leakpro.utils.logger import add_file_handler, logger
 
+modality_extensions = {"tabular": TabularExtension,
+                       "image":None,
+                       "text":None,
+                       "graph":None}
 
 class LeakPro:
     """Main class for LeakPro."""
 
-    def __init__(self:Self, handler:AbstractInputHandler, configs_path:str) -> None:
+    def __init__(self:Self, handler_class:AbstractInputHandler, configs_path:str) -> None:
 
-        assert issubclass(handler, AbstractInputHandler), "handler must be an instance of AbstractInputHandler"
+        assert issubclass(handler_class, AbstractInputHandler), "handler must be an instance of AbstractInputHandler"
```

Comment:
Really like this way of coding, more checks in runtime and you can get them for free in testing aswell. Maybe worth considering to extend this practise?

Reply:
I think we follow this quite a bit throughout but we do not commonly use asserts but exceptions.